src/components/UserInfo.js

```diff
@@ -1,38 +1,58 @@
-/**
- * "The UserInfo class is responsible for rendering information about the user on the page."
- */
 export default class UserInfo{
   #nameElement;
   #descriptionElement;
+  #avatarElement;
   #name;
   #description;
+  #avatar;
+
+/**
+ * Manage getting and setting information in profile section of page
+ * @param {object} elements {nameElement, descriptionElement, avatarElement} elements containing relevant profile information
+ * @param {object} userInfo {name, about, avatar} strings of user information to popuplate profile information on page
+ */
+  constructor(elements, userInfo){
+    //elements for name, description, and avatar
+    this.#nameElement = elements.nameElement;
+    this.#descriptionElement = elements.descriptionElement;
+    this.#avatarElement = elements.avatarElement;
+
+    //store name, description, and avatar in object
+    this.#name = userInfo.name;
+    this.#description = userInfo.about;
+    this.#avatar = userInfo.avatar;
+
+    //display name, description, and avatar in page
+    this.#nameElement.textContent = this.#name;
+    this.#descriptionElement.textContent = this.#description;
+    this.#avatarElement.src = this.#avatar;
+  }
 
   /**
-   * Manage getting and setting information in profile section of page
-   * @param {element} name element for name
-   * @param {element} description element for description
+   * Gets name and description that are currently on page
+   * @returns {object} {username, description}
    */
-  constructor(name, description){
-    this.#nameElement = name;
-    this.#descriptionElement = description;
-    this.#name = this.#nameElement.textContent;
-    this.#description = this.#descriptionElement.textContent
+  getUserInfo(){
+    return {username: this.#name, description: this.#description};
   }
 
   /**
-   * @returns object containing user info
+   * Updates user information on page
+   * @param {object} newInfo {name, about} new information to add
    */
-  getUserInfo(){
-    return {name: this.#name, description: this.#description};
+  setUserInfo(newInfo){
+    this.#name = newInfo.name;
+    this.#description = newInfo.about;
+    this.#nameElement.textContent = newInfo.name;
+    this.#descriptionElement.textContent = newInfo.about;
   }
 
   /**
-   * @param {array} param0 array containing user info [0] = name, [1] = description
+   * Replaces the current avatar image with the provided link
+   * @param {string} newLink link to replacement image 
    */
-  setUserInfo({name, description}){
-    this.#name = name;
-    this.#description = description;
-    this.#nameElement.textContent = this.#name;
-    this.#descriptionElement.textContent = this.#description;
+  updateAvatar(newLink){
+    this.#avatar = newLink;
```

Comment:
Needs correction: There is no need to save the contents in the class instance, you can find it via the dom methods